services/graph/pkg/service/v0/driveitems.go

```diff
@@ -731,34 +741,71 @@ func (g Graph) DeletePermission(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func (g Graph) getPermissionByID(ctx context.Context, permissionID string) (*libregraph.Permission, *storageprovider.ResourceId, error) {
-	share, err := g.getCS3UserShareByID(ctx, permissionID)
+func (g Graph) getPermissionByID(ctx context.Context, permissionID string, itemID *storageprovider.ResourceId) (*libregraph.Permission, *storageprovider.ResourceId, error) {
+	publicShare, err := g.getCS3PublicShareByID(ctx, permissionID)
 	if err == nil {
-		permission, err := g.cs3UserShareToPermission(ctx, share)
+		permission, err := g.libreGraphPermissionFromCS3PublicShare(publicShare)
 		if err != nil {
 			return nil, nil, err
 		}
-		return permission, share.GetResourceId(), nil
+		return permission, publicShare.GetResourceId(), nil
+	}
+
+	// The id is not referencing a public share, if the itemID is referencing
+	// a spaceroot this is a space permission. Handle that next
+	if IsSpaceRoot(itemID) {
+		gatewayClient, err := g.gatewaySelector.Next()
+		if err != nil {
+			g.logger.Debug().Err(err).Msg("selecting gatewaySelector failed")
+			return nil, nil, err
+		}
+		// get space id
+		statResponse, err := gatewayClient.Stat(ctx, &storageprovider.StatRequest{Ref: &storageprovider.Reference{ResourceId: itemID}})
+		if errCode := errorcode.FromStat(statResponse, err); errCode != nil {
+			g.logger.Warn().Err(errCode).Interface("stat.res", statResponse).Msg("stat failed")
+			return nil, nil, errCode
+		}
+		perms, err := g.getSpaceRootPermissions(ctx, statResponse.GetInfo().GetSpace().GetId())
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, p := range perms {
+			if p.GetId() == permissionID {
+				return &p, itemID, nil
+			}
+		}
 	}
 
 	var errcode errorcode.Error
 	if errors.As(err, &errcode) && errcode.GetCode() == errorcode.ItemNotFound {
-		// there is no user share with that id, check if this is a public link
-		publicShare, err := g.getCS3PublicShareByID(ctx, permissionID)
+		// there is no public link with that id, check if this is a user share
+		share, err := g.getCS3UserShareByID(ctx, permissionID)
 		if err != nil {
 			return nil, nil, err
 		}
-		permission, err := g.libreGraphPermissionFromCS3PublicShare(publicShare)
+		permission, err := g.cs3UserShareToPermission(ctx, share, false)
 		if err != nil {
 			return nil, nil, err
 		}
-		return permission, publicShare.GetResourceId(), nil
+		return permission, share.GetResourceId(), nil
 	}
 
 	return nil, nil, err
 
 }
 
+func (g Graph) getSpaceRootPermissions(ctx context.Context, spaceID *storageprovider.StorageSpaceId) ([]libregraph.Permission, error) {
+	filters := []*storageprovider.ListStorageSpacesRequest_Filter{
+		listStorageSpacesIDFilter(spaceID.GetOpaqueId()),
+	}
+	res, err := g.ListStorageSpacesWithFilters(ctx, filters, true)
```

Comment:
There is also `utils.GetSpace`. Just saying....

Reply:
Hm, nice²